elyra/pipeline/parser.py

```diff
@@ -30,37 +31,38 @@ def __init__(self, root_dir="", **kwargs):
         super().__init__(**kwargs)
         self.root_dir = root_dir
 
-    def parse(self, pipeline_definitions: Dict) -> Pipeline:
+    def parse(self, pipeline_json: Dict) -> Pipeline:
         """
         The pipeline definitions allow for defining multiple pipelines  in one json file.
         When super_nodes are used, their node actually references another pipeline in the
         set of pipeline definitions - which is "flattened" into the overall pipeline object's
         list of operations.
         """
 
-        primary_pipeline_id = pipeline_definitions['primary_pipeline']
-        primary_pipeline = PipelineParser._get_pipeline_definition(pipeline_definitions, primary_pipeline_id)
+        pipeline_definition = PipelineDefinition(pipeline_definition=pipeline_json)
+
+        primary_pipeline = pipeline_definition.primary_pipeline
 
         # runtime info is only present on primary pipeline...
-        runtime = PipelineParser._get_app_data_field(primary_pipeline, 'runtime')
+        runtime = primary_pipeline.runtime
         if not runtime:
             raise ValueError("Invalid pipeline: Missing runtime.")
-        runtime_config = PipelineParser._get_app_data_field(primary_pipeline, 'runtime-config')
+        runtime_config = primary_pipeline.runtime_config
         if not runtime_config:
             raise ValueError("Invalid pipeline: Missing runtime configuration.")
 
-        source = PipelineParser._get_app_data_field(primary_pipeline, 'source')
+        source = primary_pipeline.source
 
-        pipeline_object = Pipeline(id=primary_pipeline_id,
-                                   name=PipelineParser._get_app_data_field(primary_pipeline, 'name', 'untitled'),
+        pipeline_object = Pipeline(id=primary_pipeline.id,
```

Comment:
Here we go.  Which `Pipeline` is this?  (I know it's the one in `pipeline.py` and not in `pipeline_definition.py`, but we now have another ambiguity.)

Reply:
What does _this_ `Pipeline` value object have that the `PropertyDefinition.primary_pipeline` value object doesn't have?

I can see there being a difference for `Operations` to `Nodes` but perhaps we could merge the Operation functionality into Nodes or let the single `Pipeline` instance have runtime-representations of Nodes as a parallel list of Operations?  (Or have a Node hold its `Operation` instance?)

Reply:
I would like to consider merging these, but as another pr/initiative